src/server.js

```diff
@@ -90,32 +99,30 @@ async function sendCoins(smsIn, numberMapping) {
       const numberFormated = phoneUtil.format(number, PNF.E164);
       const result = await lookupOrCreateAddressByNumber(numberFormated);
       toAddress = result.get('address');
+
     }
   } catch(err) {
     console.error('sendCoins parse number error', JSON.stringify(err));
   }
 
   try {
-    const result = await rpc.getaddressbalance(fromAddress);
-    const balance = result.balance / 100000000;
-    const remainingBalance = balance - amount;
+    const response = await blockchain.getBalance(fromAddress);
+    const balance = response.result;
+    const remainingBalance = balance - amount - 0.0001;
 
     if (remainingBalance < 0) {
       throw Error(`Not enough coins to process your request. Balance: ${balance} BTCZ`);
     }
 
-    const transactions = [
-      { "address": toAddress, "amount": amount }
-    ];
+    const unspentOutputs = await blockchain.listunspent(fromAddress);
+    const createTx = blockchain.createTransaction;
+    const tx = createTx(unspentOutputs.result, toAddress, amount, 0.0001, numberMapping.get('WIF'));
+    const broadcast = await blockchain.broadcastTransaction(tx);
 
-    if (remainingBalance > 0) {
-      transactions.push({ "address": fromAddress, "amount": remainingBalance });
-    }
-
-    await rpc.z_sendmany(fromAddress, transactions, 1, 0);
     sendResponse(smsIn.From, `${amount} BTCZ has been sent to ${toAddress}`);
+    console.log('sendCoins done', JSON.stringify(broadcast));
   } catch(err) {
-    console.error('sendCoins error', JSON.stringify(err));
+    console.error('sendCoins error : ' + err);
```

Comment:
This should be written like this:
```
console.error('sendCoins error', { err });
```
This lets console.error and console.log print out the full json object. Otherwise, this will probably throw an error.

Reply:
Corrected all console.log to use variable literal and removed the JSON.stringify().
Also added 'console.log(`sendResponse to ${toNumber}: ${text}`);' to have a more detailed log print.